lib/signaling/room.js

```diff
@@ -141,4 +165,42 @@ class RoomSignaling extends StateMachine {
  * @param {RemoteParticipantSignaling} participantSignaling
  */
 
+/**
+ * @event RoomSignaling#event:mediaConnectionStateChanged
+ */
+
+/**
+ * @event RoomSignaling#event:signalingConnectionStateChanged
+ */
+
+/**
+ * Maybe update the {@link RoomSignaling} state.
+ * @param {RoomSignaling} roomSignaling
+ */
+function maybeUpdateState(roomSignaling) {
```

Comment:
Hmm, I wonder if the special cases could be tracked via flags as well? I imagine the logic would be similar, just checked on `get` rather than computed on state change.

Reply:
Sorry, I don't understand or can't visualize how that would work. Perhaps you could do some mutation in the `get`, but I think it's generally best to avoid that pattern? Also, you'd have to kick off whatever computation the `get` encapsulates in response to the underlying state changes. I'm not sure it's a good fit.

Reply:
Also, when you say " I wonder if the special cases could be tracked via flags as well?" — how would that be different than the current use of the flag `_mediaConnectionIsReconnecting`?

Reply:
The `get` would not do any mutation, nor would it need to be called in response to state changes. The concept here is that both our media stream and signaling classes already have concrete state machines, and rather than implementing a third state machine for `Room`, we should treat `Room.state` as a view that translates the underlying state machines' states into a singular state that is relevant to the `Room`.

The getter should not mutate anything; it would check: media state, signaling state, † and †† and output a room state.